six/modules/c++/six.sicd/tests/test_add_additional_des.cpp

```diff
@@ -0,0 +1,252 @@
+/* =========================================================================
+* This file is part of six.sidd-c++
+* =========================================================================
+*
+* (C) Copyright 2004 - 2016, MDA Information Systems LLC
+*
+* six-c++ is free software; you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation; either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with this program; If not,
+* see <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include <six/NITFWriteControl.h>
+#include <six/Types.h>
+#include <six/XMLControlFactory.h>
+#include <six/sicd/Utilities.h>
+
+
+void validateArguments(int argc, char** argv)
+{
+    if (argc != 2)
+    {
+        std::string message = "Usage: " + sys::Path::basename(argv[0]) +
+            " <XML pathname>";
+        throw except::Exception(Ctxt(message));
+    }
+    if (!str::endsWith(argv[1], ".xml"))
+    {
+        std::string message = "Extension suggests input may not be an XML file. "
+            "Expecting '*.xml'.";
+        throw except::Exception(Ctxt(message));
+    }
+}
+
+sys::Int16_T* generateBandData(const six::Data& data)
+{
+    size_t length = data.getNumRows() * data.getNumCols();
+    six::UByte* bandData =
+        new six::UByte[length * data.getNumBytesPerPixel()];
+
+    for (size_t ii = 0; ii < length; ++ii)
```

Comment:
Would it work to consistently used six:UByte instead? I’m kind of worried about getting weird errors if getNumBytesPerPixel() is other than 2.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Tuesday, May 24, 2016 8:08 PM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [ngageoint/six-library] Add method to allow adding extra DES to NITFWriteControl (#76)


In six/modules/c++/six.sicd/tests/test_add_additional_des.cpp<http://cp.mcafee.com/d/avndy0sd6Qm764mm7xNEVdTdIIczxNEVvvdLzCm6kjhOqejqdTHLe6zCVEVudEEIEFLCzBNxYS8RxIOuMVnZxg5aEqTjBPqG6JQVsSWxrS61P_nV6VEVjsKzRXBQnC7Kjjsj7fLZh5dqWqJQnel3PWApmU6CQjqpK_cI9TKY_txcTsSjDdqymohBcCSgVv84t8_mk9VigunfAgzIiSbdDDUes0N2lbdQFaI1mbz31EVsso76UlzWNVo54pj9JAendLITjdwLQzh0qmU4t8_mk9VigunfAgzIiNFtd40wwwq85vCjGwq89Rd46DDRc6y3k49g8Cy3M-std40sd31EwDkQg0Md18K0IpjYiH2sqDCy0Qk0oesd59IQsTdRYCE>:

> +    }

> +    if (!str::endsWith(argv[1], ".xml"))

> +    {

> +        std::string message = "Extension suggests input may not be an XML file. "

> +            "Expecting '*.xml'.";

> +        throw except::Exception(Ctxt(message));

> +    }

> +}

> +

> +sys::Int16_T* generateBandData(const six::Data& data)

> +{

> +    size_t length = data.getNumRows() * data.getNumCols();

> +    six::UByte* bandData =

> +        new six::UByte[length * data.getNumBytesPerPixel()];

> +

> +    for (size_t ii = 0; ii < length; ++ii)

Why not just treat it as a sys::Int16_T all the way through? You can cast ii to that instead...

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<http://cp.mcafee.com/d/avndy0Q83gOrhosohpou76zATsSOMOe76zBZYS-epophd79EVdETuKYUqerCzBUSyyOyC-qen67Pozm6P9X3BvS50kGxHtendGEqTjBPrG5Loo7fZvArCzBdOWfnKnhuouVddNcs-_R4kRHFGThsVkffGhBrwqrjdFCXYOMDuXPZS4PtPpesRG9px6kOrp3BYwhQzZpgDB91Vs-h2eNboISuvwVP7p7_j25PhODtVwseIevghj8c_tMQsLL-VWa3XMVTTbISvKy1An663hOUUMedMH7RzOMa8OCjr8sKrvpKCr1vF6y0QJM8Wh-IEjOAwYKv8x7oBziWq81110Qga_cDl0QgjGq8dffGod46E8iwhd47xYUWq80Uq63h1eFEw1wq2hs1oODUBm4URfd41EE0MsUqajpEVKrYt9Z>


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.